internal/controller/utils/sync/nfspvc.go

```diff
@@ -70,7 +70,9 @@ func handlePvState(ctx context.Context, nfspvc danaiov1alpha1.NfsPvc, log logr.L
 	}
 
 	if pv.Status.Phase == corev1.VolumeReleased || pv.Status.Phase == corev1.VolumeFailed ||
-		(nfspvc.Status.PvPhase == string(corev1.VolumeBound) && nfspvc.Status.PvcPhase == string(corev1.ClaimPending)) {
+		(nfspvc.Status.PvPhase == string(corev1.VolumeBound) &&
+			nfspvc.Status.PvcPhase == string(corev1.ClaimPending) &&
+			verifyPvcUid(ctx, k8sClient, pv.Spec.ClaimRef.UID, nfspvc)) {
```

Comment:
The `if` here is very long and it's difficult to understand exactly what it checks. I would put these checks in an external function with a relevant name and a comment above it.